app/src/main/java/nl/tudelft/b_b_w/database/Database.java

```diff
@@ -0,0 +1,189 @@
+    static final int INDEX_PREV_HASH_SENDER = 5;
+    static final int INDEX_REVOKE = 6;
+    static final int INDEX_TRUST_VALUE = 7;
+
+    /**
+     * Column names for users
+     */
+    // KEY_OWNER is the same as blocks table
+    static final String KEY_NAME = "name";
+    static final String KEY_IBAN = "iban";
+
+    /**
+     * Column indices for users
+     */
+    static final int INDEX_NAME = 0;
+    static final int INDEX_IBAN = 1;
+    static final int INDEX_PUBLICKEY = 2;
+
+    /**
+     * Columns in the user table
+     */
+    static final String[] USER_COLUMNS = new String[] {
+            KEY_NAME, KEY_IBAN, KEY_CONTACT
+    };
+
+    /**
+     * Our version is one
+     */
+    static final int DATABASE_VERSION = 1;
+
+    /**
+     * How our database is called
+     */
+    static final String DATABASE_NAME = "blockChain";
+
+    /**
+     * Create a new connection to the database
+     * @param context the specific context containing our database
+     */
+    public Database(Context context) {
+        super(context, DATABASE_NAME, null, DATABASE_VERSION);
+    }
+
+    /**
+     * Called when the database is created for the first time. This is where the
+     * creation of tables and the initial population of the tables should happen.
+     *
+     * @param db The database.
+     */
+    @Override
+    public void onCreate(SQLiteDatabase db) {
+        db.execSQL(createBlockTable);
+        db.execSQL(createUserTable);
+    }
+
+    /**
+     * Called when the database needs to be upgraded. The implementation
+     * should use this method to drop tables, add tables, or do anything else it
+     * needs to upgrade to the new schema version.
+     * <p>
+     * <p>
+     * The SQLite ALTER TABLE documentation can be found
+     * <a href="http://sqlite.org/lang_altertable.html">here</a>. If you add new columns
+     * you can use ALTER TABLE to insert them into a live table. If you rename or remove columns
+     * you can use ALTER TABLE to rename the old table, then create the new table and then
+     * populate the new table with the contents of the old table.
+     * </p><p>
+     * This method executes within a transaction.  If an exception is thrown, all changes
+     * will automatically be rolled back.
+     * </p>
+     *
+     * @param db         The database.
+     * @param oldVersion The old database version.
+     * @param newVersion The new database version.
+     */
+    @Override
+    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+        final String upgradeScript = "DROP TABLE IF EXISTS " + BLOCK_TABLE_NAME + ";"
+                + "DROP TABLE IF EXISTS option;";
+
+        // TODO: check if the db version is lower than the latest
+
+        db.execSQL(upgradeScript);
+    }
+
+    /**
+     * Perform a read query
+     * @param query the query to execute
+     * TODO: think about if we need to open a new connection every time if it is too slow
```

Comment:
The scalability is not harmed since the database connection closing has nothing to do with multithreading: it can also be done in multiple threads at once